docs/tutorials/interceptors.md

```diff
@@ -0,0 +1,109 @@
+# Interceptors
+
+## Introduction
+
+Interceptors can be seen as a middleware that modifies incoming or outgoing requests according to a specific logic. In AI DIAL, we use interceptors to facilitate the implementation of a so-called Responsible AI approach and enforce compliance with internal and external privacy regulations and policies.
+
+For example, interceptors can block requests that violate specific regulations, related to restricted domains, or potentially lead to data leaks or biased responses. Another use case is when interceptors allow applications or models to respond solely to specific subjects and anonymize Personally Identifiable Information (PII) from user requests, or cache LLM responses.
+
+Technically speaking, interceptors in AI DIAL are components inserted into deployments (applications or model adapters) that can be called before or after [chat completion requests](https://epam-rail.com/dial_api#/paths/~1openai~1deployments~1%7BDeployment%20Name%7D~1chat~1completions/post).
+
+Interceptors in AI DIAL could be classified into the following categories:
+
+* **Pre-interceptors** that only modify the incoming request from the client (e.g. rejecting requests following certain criteria)
+* **Post-interceptors** that only modify the response received from the upstream (e.g. censoring the response)
+* **Generic interceptors** that modify both the incoming request and the response from the upstream (e.g. caching the responses)
+
+For example, to implement PII (Personally Identifiable Information) anonymization for all data sent to models through AI DIAL, you can use a **generic** interceptor which can employ specific locally deployed NLP models to obfuscate (replace with token) PII in requests (pre-interceptor) and decode it in responses (post-interceptor), effectively ensuring the anonymization of all personal data.
+
+<!-- Interceptors in AI DIAL can:
+
+* Modify an incoming AI DIAL request received from the client (or it may leave it as is).
+* Call an upstream AI DIAL application (the upstream for short) with the modified request.
+* Modify the response from the upstream (or it may leave it as is).
+* Return the modified response to the client. -->
+
+The stack of interceptors in AI DIAL works similarly to a stack of middleware in Express.js or Django. Every request/response in the diagram above goes through AI DIAL Core (this is hidden from the diagram for brevity):
```

Comment:
The sentence refers to the stacks but doesn't contains any links to examples. That may confuse a reader. I would suggest adding links to the stacks or re-phrase the sentence not mentioning any stacks.